lib/Server.js

```diff
@@ -42,6 +42,11 @@ class Server {
   constructor(compiler, options = {}) {
     validate(schema, options, 'webpack Dev Server');
 
+    if (compiler.devServer) {
+      throw new Error('The compiler already has a dev server attached.');
+    }
+    compiler.devServer = this;
```

Comment:
Potential memory leak, we need to use `Symbol` to detect

Reply:
@alexander-akait Which do you mean by using `Symbol`:
```js
compiler[Symbol.for('devServer')] = this;
```
or
```js
compiler[Symbol.for('hasDevServer')] = true;
```

Reply:
better webpack-dev-server

Reply:
@alexander-akait Should I store only a boolean value, since the reference to the server instance would be the one causing a potential memory leak?

Reply:
Yep, let's keep boolean value